Account/FirefoxAccount.swift

```diff
@@ -140,21 +149,26 @@ public class FirefoxAccount {
             configurationLabel = configurationLabel,
             email = dictionary["email"] as? String,
             uid = dictionary["uid"] as? String {
+                let fxaDeviceId = dictionary["fxaDeviceId"] as? String
+                let deviceRegistrationVersion = dictionary["deviceRegistrationVersion"] as? Int ?? 0
                 let stateCache = KeychainCache.fromBranch("account.state", withLabel: dictionary["stateKeyLabel"] as? String, withDefault: SeparatedState(), factory: stateFromJSON)
                 return FirefoxAccount(
                     configuration: configurationLabel.toConfiguration(),
                     email: email, uid: uid,
+                    fxaDeviceId: fxaDeviceId, deviceRegistrationVersion: deviceRegistrationVersion,
                     stateCache: stateCache)
         }
         return nil
     }
 
     public enum AccountError: MaybeErrorType {
         case NotMarried
+        case DeviceRegistrationFailed
```

Comment:
I think of device registration as an identity-attached service, just like Sync. It's possible to do FxA-ey things without being a registered device. One of the discriminant questions here: if the device registration endpoint is down, should we consider FxA signin to have failed? I don't think so, which implies that it's separate. See `SyncStateMachine` for one kind of chaining concept here.